lib/ansible/modules/cloud/vmware/vmware_host_facts.py

```diff
@@ -0,0 +1,178 @@
+#!/usr/bin/python
+# -*- coding: utf-8 -*-
+
+# (c) 2017, Wei Gao <gaowei3@qq.com>
+#
+# GNU General Public License v3.0+ (see COPYING or https://www.gnu.org/licenses/gpl-3.0.txt)
+
+from __future__ import absolute_import, division, print_function
+__metaclass__ = type
+
+ANSIBLE_METADATA = {'metadata_version': '1.1',
+                    'status': ['preview'],
+                    'supported_by': 'community'}
+
+
+DOCUMENTATION = '''
+---
+module: vmware_host_facts
+short_description: Gathers facts about remote vmware host
+description:
+    - Gathers facts about remote vmware host
+version_added: 2.5
+author:
+    - Wei Gao (@woshihaoren)
+notes:
+    - Tested on vSphere 5.5
+requirements:
+    - "python >= 2.6"
+    - PyVmomi
+extends_documentation_fragment: vmware.documentation
+'''
+
+EXAMPLES = '''
+- name: Gather vmware host facts
+  local_action:
+    module: vmware_host_facts
+    hostname: esxi_ip_or_hostname
+    username: username
+    password: password
+'''
+
+RETURN = '''
+instance:
+    description: system info about the host machine
+    returned: always
+    type: dict
+    sample: None
```

Comment:
If there's no sample, please remove that line.

Reply:
sample is 
    "ansible_facts": {
        "ansible_all_ipv4_addresses": [
            "10.255.22.105"
        ],
        "ansible_bios_date": "2014-07-09T00:00:00+00:00",
        "ansible_bios_version": "2.4.3",
        "ansible_datastore": [
            {
                "free": "1.1TB",
                "name": "datastore-22.105",
                "total": "5.4TB"
            }
        ],
        "ansible_distribution": "VMware ESXi",
        "ansible_distribution_build": "1331820",
        "ansible_distribution_version": "5.5.0",
        "ansible_hostname": "VM-111",
        "ansible_interfaces": [
            "vmk0"
        ],
        "ansible_memfree_mb": 4261,
        "ansible_memtotal_mb": 65490,
        "ansible_os_type": "vmnix-x86",
        "ansible_processor": "Intel(R) Xeon(R) CPU E5-2630 v2 @ 2.60GHz",
        "ansible_processor_cores": 12,
        "ansible_processor_count": 2,
        "ansible_processor_vcpus": 24,
        "ansible_product_serial": "74M4Z41",
        "ansible_product_name": "PowerEdge R720",
        "ansible_system_vendor": "Dell Inc.",
        "ansible_vmk0": {
            "device": "vmk0",
            "ipv4": {
                "address": "10.255.22.105",
                "netmask": "255.255.255.0"
            },
            "macaddress": "b0:83:fe:e6:f4:1d",
            "mtu": 1500
        }
    }

I think it's to long, so remove or add ?